### PR TITLE
Added Effect "The Blade Hungers"

### PR DESCRIPTION
**Pull request by ProfessorBiddle**
This is (sort of) from an escalation in Hitman 3 where 47 is given a katana and told "The Blade Hungers"

The screen flickers and slowly fades to red, and if it goes fully red you die. To reset the fade, you have to kill someone with the machete that the player is forced to hold.

### CODE REVIEW

**Comment by BillNyeTheRussianSpyV2** [comment]
Yes. I loved that part of the esclation

File: ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+static void OnTick()
+{
+	Ped playerPed = PLAYER_PED_ID();
+
+	for (Hash weapon : Memory::GetAllWeapons())
+	{
+		if (weapon != bladeHash)
+		{
+			REMOVE_WEAPON_FROM_PED(playerPed, weapon);
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp:29]: Why remove all weapons? You can just force equip a machete and disable weapon switching

**Review by Last0xygen — CHANGES_REQUESTED** [review]

File: ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp (hunk 1 above)
Comment (ProfessorBiddle) [reply ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp:29]: I didn't think about that at all, that's a good idea lol

Comment (Last0xygen) [reply ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp]: Now add blocking any weapon-change 🤪

Comment (ProfessorBiddle) [reply ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp]: No need, the equipnow tag forces the player to equip the machete at all times

Comment (Last0xygen) [reply ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp]: sure, but why let him? ;P yeah, but looks good to me 👍

**Comment by pongo1231** [comment]
You might want to consider making it short timed to make sure it doesn't go on for too long (or moving the logic to OnStart + a while loop and stopping the effect after a (couple of) kill(s) / player death).

**Comment by ProfessorBiddle** [comment]
>(or moving the logic to OnStart + a while loop and stopping the effect after a (couple of) kill(s) / player death).

I'll give this a try, but as a short effect it doesn't really work well as the time it takes to do anything is about 20-25 seconds

**Comment by pongo1231** [comment]
In that case I'll just merge it in its current state and then either keep or change it depending on the feedback it gets.

**Pull request closed without merging**